recipe/meta.yaml

```diff
@@ -1,51 +1,99 @@
 
 build:
-  script:
-    - {{ PYTHON }} -m pip install ./arro3-core -vv  # [arro3_module == "arro3-core"]
-    - {{ PYTHON }} -m pip install ./arro3-compute -vv  # [arro3_module == "arro3-compute"]
-    - {{ PYTHON }} -m pip install ./arro3-io -vv  # [arro3_module == "arro3-io"]
-    - cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
   number: 0
 
-requirements:
-  build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - maturin >=1.7.0,<2.0                   # [build_platform != target_platform]
-    - {{ stdlib("c") }}
-    - {{ compiler("c") }}
-    - {{ compiler("rust") }}
-    - cargo-bundle-licenses
-  host:
-    - python
-    - maturin >=1.4.0,<2.0
-    - pip
-  run:
-    - python
-    - arro3-core  # [arro3_module == "arro3-compute"]
-    - arro3-core  # [arro3_module == "arro3-io"]
-    - typing_extensions >=4.6.0  # [py<312]
+outputs:
+  - name: arro3-core
+    script: build-pkg.sh    # [not win]
+    script: build-pkg.bat   # [win]
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+        - maturin >=1.7.0,<2.0                   # [build_platform != target_platform]
+        - {{ stdlib("c") }}
+        - {{ compiler("c") }}
+        - {{ compiler("rust") }}
+        - cargo-bundle-licenses
+      host:
+        - python
+        - maturin >=1.4.0,<2.0
+        - pip
+      run:
+        - python
+        - typing_extensions >=4.6.0  # [py<312]    
+    test:
+      imports:
+        - arro3.core
+      commands:
+        - pip check
+      requires:
+        - pip
 
-test:
-  imports:
-    - arro3.core  # [arro3_module == "arro3-core"]
-    - arro3.compute  # [arro3_module == "arro3-compute"]
-    - arro3.io  # [arro3_module == "arro3-io"]
-  commands:
-    - pip check
-  requires:
-    - pip
+  - name: arro3-compute
+    script: build-pkg.sh    # [not win]
+    script: build-pkg.bat   # [win]
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+        - maturin >=1.7.0,<2.0                   # [build_platform != target_platform]
+        - {{ stdlib("c") }}
+        - {{ compiler("c") }}
+        - {{ compiler("rust") }}
+        - cargo-bundle-licenses
+      host:
+        - python
+        - maturin >=1.4.0,<2.0
+        - pip
+      run:
+        - python
+        - {{ pin_subpackage("arro3-core", exact=True) }}
+        - typing_extensions >=4.6.0  # [py<312]    
```

Comment:
This is really cool. I wasn't particularly familiar with how to set up these more complex conda forge recipes.

Reply:
I still don't fully understand how the `run` dependency can find `arro3-core` (and seems like that's tested in the `arro3-compute` import?)

Reply:
conda-build recipes can build multiple outputs, i.e. multiple conda packages in the same recipe. The packages are built in an order determined by conda-build to satisfy the dependency resolution constraints. So, conda-build first builds the output `arro3-core` and then builds the other two `aro3-compute` and `arro3-io` in some order.